syntax = "proto3";

package MatrixMult;

// matrix service definition
service MatrixService{
  rpc MultiplyBlock (MatrixRequest) returns (MatrixReply) {}
  rpc AddBlock (MatrixRequest) returns (MatrixReply) {}
}

// request message containing the two matrices
message MatrixRequest {
  int32 a00 = 1;
  int32 a01 = 2;
  int32 a10 = 3;
  int32 a11 = 4;
  int32 b00 = 5;
  int32 b01 = 6;
  int32 b10 = 7;
  int32 b11 = 8;
}

// response message containing the result matrix
message MatrixReply {
  int32 c00 = 1;
  int32 c01 = 2;
  int32 c10 = 3;
  int32 c11 = 4;
}